### ISSUE TITLE
numpy.linalg.LinAlgError: 2-th leading minor of the array is not positive definite

### ISSUE THREAD

**Issue by prashant-py-debug** [mikel-brostrom/Yolov5_DeepSort_Pytorch#231]
for some video it works fine but for some after running for a while it gives this error. I am also resizing the videos to (640,480). please help.

covariance: [[     10.112           0           0           0      2.4077           0           0           0]
 [          0           0           0           0           0           0           0           0]
 [          0           0      14.913           0           0           0    0.074192           0]
 [          0           0           0      14.156           0           0           0      3.3704]
 [     2.4077           0           0           0      2.4318           0           0           0]
 [          0           0           0           0           0           0           0           0]
 [          0           0    0.074192           0           0           0     0.14838           0]
 [          0           0           0      3.3704           0           0           0      3.4041]]
Traceback (most recent call last):
  File "C:\Third_eye_approach3_yolov5\tracking\Yolov5_DeepSort_Pytorch\custom.py", line 92, in <module>
    outputs = deepsort.update(xywhs, confs, clss, frame)
  File "C:\Third_eye_approach3_yolov5\tracking\Yolov5_DeepSort_Pytorch\deep_sort_pytorch\deep_sort\deep_sort.py", line 39, in update
    self.tracker.update(detections, classes)
  File "C:\Third_eye_approach3_yolov5\tracking\Yolov5_DeepSort_Pytorch\deep_sort_pytorch\deep_sort\sort\tracker.py", line 78, in update
    self.tracks[track_idx].update(
  File "C:\Third_eye_approach3_yolov5\tracking\Yolov5_DeepSort_Pytorch\deep_sort_pytorch\deep_sort\sort\track.py", line 155, in update
    self.mean, self.covariance = kf.update(
  File "C:\Third_eye_approach3_yolov5\tracking\Yolov5_DeepSort_Pytorch\deep_sort_pytorch\deep_sort\sort\kalman_filter.py", line 175, in update
    chol_factor, lower = scipy.linalg.cho_factor(
  File "C:\Users\prash\AppData\Local\Programs\Python\Python39\lib\site-packages\scipy\linalg\decomp_cholesky.py", line 152, in cho_factor
    c, lower = _cholesky(a, lower=lower, overwrite_a=overwrite_a, clean=False,
  File "C:\Users\prash\AppData\Local\Programs\Python\Python39\lib\site-packages\scipy\linalg\decomp_cholesky.py", line 37, in _cholesky
    raise LinAlgError("%d-th leading minor of the array is not positive "
numpy.linalg.LinAlgError: 2-th leading minor of the array is not positive definite

**Comment by ppalasek** [comment]
Hi. I just tried using the implementation from this repository today (thank you @mikel-brostrom) and faced the same issue. I noticed that the first two elements of bbox_xywh passed to the deepsort.update function should represent the x, y coordinates of the centre of the bounding box, not the top left corner.

By changing the rows in bbox_xywh from [x, y, w, h] to [x + w / 2, y + h / 2] the error disappeared.

This also fixed the alignment of the bounding boxes returned from the update function, I noticed they were shifted when plotting them previously.

Best,
Petar
